cli/src/cpp/utils/qos_parser.cpp

```diff
@@ -40,96 +45,196 @@ void qos_parser(
+                                    if (values[DEFAULT_POSITION] == CLI_PERSISTENT_ARGUMENT)
+                                    {
+                                        durability_kind = LIB_PERSISTENT_ARGUMENT;
+                                    }
+                                    else if (values[DEFAULT_POSITION] == CLI_TRANSIENT_ARGUMENT)
+                                    {
+                                        durability_kind = LIB_TRANSIENT_ARGUMENT;
+                                    }
+                                    else if (values[DEFAULT_POSITION] == CLI_TRANSIENT_LOCAL_ARGUMENT)
+                                    {
+                                        durability_kind = LIB_TRANSIENT_LOCAL_ARGUMENT;
+                                    }
+                                    else if (values[DEFAULT_POSITION] == CLI_VOLATILE_ARGUMENT)
+                                    {
+                                        durability_kind = LIB_VOLATILE_ARGUMENT;
+                                    }
+                                    else
+                                    {
+                                        durability_kind = values[DEFAULT_POSITION];
+                                    }
+
+                                    qosprof::data_reader::qos::set_durability_kind(filename, profile_name,
+                                            durability_kind);
+                                }
+                                break;
+                        }
+                        break;
+                    case DDSEntity::DATAWRITER:
+                        switch (command)
+                        {
+                            case CommonCommands::CLEAR:
+                                print_usage = !check_command_arguments(command, 0, values.size(), message.str(), true);
+                                if (!print_usage)
+                                {
+                                    // TODO
+                                }
+                                break;
+                            case CommonCommands::PRINT:
+                                print_usage = !check_command_arguments(command, 0, values.size(), message.str(), true);
+                                if (!print_usage)
+                                {
+                                    // TODO
+                                }
+                                break;
+                            case CommonCommands::QUERY:
+                                print_usage = !query_not_allowed(message.str());
+                                break;
+                            case CommonCommands::SET:
+                                print_usage = !check_command_arguments(command, 1, values.size(), message.str(), true);
+                                if (!print_usage)
+                                {
+                                    qosprof::data_writer::qos::set_durability_kind(filename, profile_name,
```

Comment:
Shall the given kind be checked also here, as it has been checked in the `DataReader`?

Reply:
You are right. I am going to refactor to check first the command so this piece of code is written only once (instead of wrapping it in an auxiliary function)